src/pymor/algorithms/greedy.py

```diff
@@ -7,22 +7,149 @@
 from pymor.core.logger import getLogger
 from pymor.parallel.dummy import dummy_pool
-from pymor.parallel.manager import RemoteObjectManager
+from pymor.parallel.interfaces import RemoteObjectInterface
+from pymor.tools.deprecated import Deprecated
 
 
-def greedy(fom, reductor, samples, use_estimator=True, error_norm=None,
-           atol=None, rtol=None, max_extensions=None, extension_params=None, pool=None):
-    """Greedy basis generation algorithm.
+def weak_greedy(surrogate, training_set, atol=None, rtol=None, max_extensions=None, pool=None):
+    """Weak greedy basis generation algorithm [BCDDPW11]_.
 
-    This algorithm generates a reduced basis by iteratively adding the
-    worst approximated solution snapshot for a given training set to the
-    reduced basis. The approximation error is computed either by directly
-    comparing the reduced solution to the detailed solution or by using
-    an error estimator (`use_estimator == True`). The reduction and basis
-    extension steps are performed by calling the methods provided by the
-    `reductor` and `extension_algorithm` arguments.
+    This algorithm generates an approximation basis for a given set vectors
+    associated with a training set of parameters by iteratively evaluating a
+    :class:`surrogate <WeakGreedySurrogate>` for the approximation error on
+    the training set and adding the worst approximated vector (according to
+    the surrogate) to the basis.
+
+    The constructed basis is extracted from the surrogate after termination
+    of the algorithm.
+
+    Parameters
+    ----------
+    surrogate
+        An instance of :class:`WeakGreedySurrogate` representing the surrogate
+        for the approximation error.
+    training_set
+        The set of parameter samples on which to perform the greedy search.
+    atol
+        If not `None`, stop the algorithm if the maximum (estimated) error
+        on the training set drops below this value.
+    rtol
+        If not `None`, stop the algorithm if the maximum (estimated)
+        relative error on the training set drops below this value.
+    max_extensions
+        If not `None`, stop the algorithm after `max_extensions` extension
+        steps.
+    pool
+        If not `None`, a |WorkerPool| to use for parallelization. Parallelization
+        needs to be supported by `surrogate`.
+
+    Returns
+    -------
+    Dict with the following fields:
+
+        :max_errs:               Sequence of maximum estimated errors during the greedy run.
+        :max_err_mus:            The parameters corresponding to `max_errs`.
+        :extensions:             Number of performed basis extensions.
+        :time:                   Total runtime of the algorithm.
+    """
+    logger = getLogger('pymor.algorithms.greedy.weak_greedy')
+    training_set = list(training_set)
+    logger.info(f'Started greedy search on training set of size {len(training_set)}')
```

Comment:
Not trying to be picky, but you finish most other log messages with a trailing `.`...

Reply:
Fixed.